sdk/python/foundation-models/system/inference/text-to-image/text-to-image-inpainting-batch-endpoint.ipynb

```diff
@@ -0,0 +1,507 @@
+   "outputs": [],
+   "source": [
+    "# Read all the csvs in the data folder into a pandas dataframe\n",
+    "import glob\n",
+    "import os\n",
+    "import pandas as pd\n",
+    "\n",
+    "# Specify the folder where your CSV files are located\n",
+    "dataset_parent_dir = \"inpainting_data/batch_data\"\n",
+    "\n",
+    "# Use glob to get a list of CSV files in the folder\n",
+    "csv_files = glob.glob(os.path.join(dataset_parent_dir, \"*.csv\"))\n",
+    "\n",
+    "# Read all CSV files into a single DataFrame using pd.concat\n",
+    "batch_df = pd.concat((pd.read_csv(file) for file in csv_files), ignore_index=True)\n",
+    "\n",
+    "# Now, 'combined_df' contains all the data from the CSV files in the folder\n",
+    "print(batch_df.head())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from pathlib import Path\n",
+    "\n",
+    "# Specify the folder where your CSV files should be saved\n",
+    "processed_dataset_parent_dir = \"inpainting_data/processed_batch_data\"\n",
+    "os.makedirs(processed_dataset_parent_dir, exist_ok=True)\n",
+    "batch_input_file = \"batch_input.csv\"\n",
+    "\n",
+    "# Divide this into files of <x> rows each\n",
+    "batch_size_per_predict = 2\n",
+    "for i in range(0, len(batch_df), batch_size_per_predict):\n",
+    "    j = i + batch_size_per_predict\n",
+    "    batch_df[i:j].to_csv(os.path.join(processed_dataset_parent_dir, str(i) + batch_input_file))\n",
+    "\n",
+    "# Check out the first and last file name created\n",
+    "input_paths = sorted(Path(processed_dataset_parent_dir).iterdir(), key=os.path.getmtime)\n",
+    "input_files = [os.path.basename(path) for path in input_paths]\n",
+    "print(f\"{input_files[0]} to {str(i)}{batch_input_file}.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Register folder containing csv files in AML as data asset to use in batch job."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "input = Input(path=processed_dataset_parent_dir, type=AssetTypes.URI_FOLDER)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 4. Deploy the model to a batch endpoint\n",
+    "Batch endpoints are endpoints that are used to do batch inferencing on large volumes of data over a period of time. The endpoints receive pointers to data and run jobs asynchronously to process the data in parallel on compute clusters. Batch endpoints store outputs to a data store for further analysis. For more information on batch endpoints and deployments, see <a href=\"https://learn.microsoft.com/en-us/azure/machine-learning/concept-endpoints?view=azureml-api-2#what-are-batch-endpoints\" target=\"_blank\"> What are batch endpoints?</a>. In this sub-section, we will cover the following items.\n",
```

Comment:
What are batch endpoints?.

- Need space before ?
- ? should not be followed with period